src/MongoDB.Driver.Core/Core/Compression/SnappyCompressor.cs

```diff
@@ -35,17 +32,19 @@ internal class SnappyCompressor : ICompressor
         /// <param name="output">The output stream.</param>
         public void Compress(Stream input, Stream output)
         {
-#if NET452 || NETSTANDARD2_0
             var uncompressedSize = (int) (input.Length - input.Position);
             var uncompressedBytes = new byte[uncompressedSize]; // does not include uncompressed message headers
             input.ReadBytes(uncompressedBytes, offset: 0, count: uncompressedSize, CancellationToken.None);
             var maxCompressedSize = SnappyCodec.GetMaxCompressedLength(uncompressedSize);
             var compressedBytes = new byte[maxCompressedSize];
-            var compressedSize = SnappyCodec.Compress(uncompressedBytes, 0, uncompressedSize, compressedBytes, 0);
-            output.Write(compressedBytes, 0, compressedSize);
-#else
-            throw new NotSupportedException();
-#endif
+            var compressedSize = SnappyCodec.Compress(
+                input: uncompressedBytes, 
+                inputOffset: 0, 
+                inputLength: uncompressedSize, 
+                output: compressedBytes, 
+                outputOffset: 0, 
+                outputLength: compressedBytes.Length); // output.Length - outputOffset
```

Comment:
I understood the idea. We can do suggested change. It wasn't clear for me at first glance, because I didn't notice that passing `compressedBytes.Length` will give the same result as the internal formula to calculate `outputLength`:

     output.Length - outputOffset => compressedBytes.Length - 0 = compressedBytes.Length